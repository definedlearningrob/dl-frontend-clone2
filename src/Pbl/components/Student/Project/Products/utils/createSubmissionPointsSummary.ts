export const createSubmissionPointsSummary = (pointsAvailable: number, pointsScored: number) =>
  `${pointsScored}/${pointsAvailable}`;
