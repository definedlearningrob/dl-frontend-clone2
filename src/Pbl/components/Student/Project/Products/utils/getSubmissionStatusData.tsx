import { capitalize } from 'lodash-es';

import { PRODUCT_SUBMISSION_STATUS } from '@pbl/resources/enums';

import { SUBMISSION_STATUS } from '@shared/components/SubmissionStatusBadge';

export const getSubmissionStatusData = (
  status: PRODUCT_SUBMISSION_STATUS,
  gradePoints?: string | null
) => {
  const statusLabel = capitalize(status).replace(/[_]/g, ' ');
  const label = [statusLabel, gradePoints].filter(Boolean).join(' ');

  switch (status) {
    case PRODUCT_SUBMISSION_STATUS.DRAFT:
      return { label, value: SUBMISSION_STATUS.DRAFT };
    case PRODUCT_SUBMISSION_STATUS.SUBMITTED:
      return {
        label,
        value: SUBMISSION_STATUS.SUBMITTED,
      };
    case PRODUCT_SUBMISSION_STATUS.GRADED:
      return { label, value: SUBMISSION_STATUS.GRADED };
    case PRODUCT_SUBMISSION_STATUS.NOT_STARTED:
      return {
        label,
        value: SUBMISSION_STATUS.NOT_STARTED,
      };
  }
};
