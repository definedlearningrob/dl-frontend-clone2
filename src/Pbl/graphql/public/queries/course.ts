import { gql } from '@apollo/client';

import COURSE_DETAILS_FIELDS, { TCourseData } from '@pbl/graphql/fragments/course';

export default gql`
  ${COURSE_DETAILS_FIELDS}
  query PublicCourseDetails($shareId: ID!, $code: String!, $courseId: ID!) {
    project: task(shareId: $shareId, code: $code) {
      course(id: $courseId) {
        exploreMoreAvailable
        ...CourseDetails
      }
    }
  }
`;

export type TCourseDetailsData = {
  project: {
    course: TCourseData & {
      exploreMoreAvailable: boolean;
    };
  };
};

export type TCourseDetailsVariables = {
  shareId: string;
  code: string;
  courseId: string;
};
