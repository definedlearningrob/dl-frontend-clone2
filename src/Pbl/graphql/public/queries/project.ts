import { gql } from '@apollo/client';
import { PresentationTypes } from '@graphql/shared/shared/types';

import {
  TCheckInGroup,
  TCheckInQuestion,
  TDefinedCareer,
  TProduct,
} from '@pbl/components/Project/types';

import { ContentStatusesTypes } from '@shared/resources/enums';

export default gql`
  query PublicProject($shareId: ID!, $code: String!, $trackPresentation: Boolean) {
    project: task(shareId: $shareId, code: $code, track: true) {
      checkInGroups {
        displayName
        id
        name
        questions {
          id
          question
          step
        }
        step
      }
      checkInQuestions {
        id
        question
        step
      }
      courses {
        id
        name
        thumbnailUrl
        pathwayName
      }
      description
      displayName
      files {
        description
        displayName
        filename
        id
        step
        url
      }
      id
      introduction
      presentation(track: $trackPresentation) {
        color
        description
        displayName
        id
        name
        slides {
          backgroundColor
          backgroundImage
          checkInQuestions {
            id
            question
            step
          }
          checkInGroups {
            displayName
            id
            name
            step
            questions {
              id
              question
              step
            }
          }
          content {
            id
            images {
              contentId
              id
              url
              style
              thumbnailUrl
              position
            }
            links {
              targetId
              targetName
              text
              contentId
            }
            texts {
              contentId
              type
              value
              style
            }
            videos {
              id
              contentId
              url
              filename
              videoUrl
            }
          }
          description
          id
          iframeUrl
          name
          notes
          step
          template
          subslides {
            backgroundColor
            backgroundImage
            content {
              id
              images {
                contentId
                id
                url
                style
                position
              }
              links {
                targetId
                targetName
                text
                contentId
              }
              texts {
                contentId
                type
                value
                style
              }
              videos {
                id
                contentId
                url
                filename
                videoUrl
              }
            }
            description
            id
            iframeUrl
            name
            notes
            step
            template
          }
          products {
            description
            displayName
            id
            name
            rubricsUrl
            rubrics {
              description
              displayName
              id
              name
              uuid
              criterias {
                id
                rubricCriteriaLabelId
                rubricHeadingId
                text
                uuid
              }
              headings {
                id
                multiplier
                name
                uuid
              }
              criteriaLabels {
                displayName
                id
                score
                uuid
              }
            }
            step
          }
        }

        status
        transition
        typography
        type
      }
      presentationUrl
      standard
      studentResources
      teachingResources
      units {
        displayName
        id
      }
    }
  }
`;

export type TProject = {
  checkInGroups: TCheckInGroup[];
  checkInQuestions: TCheckInQuestion[];
  courses: TDefinedCareer[];
  description: string;
  displayName: string;
  files: TFile[];
  id: string;
  introduction: string;
  presentation: TProjectPresentation;
  presentationUrl: string;
  standard: string;
  studentResources: string;
  teachingResources: string;
  units: TUnit[];
};

export type TProjectPresentationSlideTemplate =
  | 'basicText'
  | 'title'
  | 'imageText'
  | 'iframe'
  | 'video'
  | 'twoProductChoice'
  | 'threeProductChoice'
  | 'fourProductChoice'
  | 'fiveProductChoice'
  | 'product'
  | 'checkInQuestion'
  | 'checkInGroup';

export type TProjectPresentationSlideTransition =
  | 'none'
  | 'fade'
  | 'slide'
  | 'convex'
  | 'concave'
  | 'zoom';

export type TProjectPresentationImageStyle = 'fill' | 'contain';
export type TTaskPresentationImagePosition = 'center' | 'left' | 'right';

export type TProjectPresentationTypography = 'lora' | 'roboto' | 'cabinSketch' | 'montserrat';

export type TProjectPresentationVideo = {
  contentId: string;
  id: string;
  url?: string;
  videoUrl?: string;
  filename?: string;
};
export type TProjectPresentationImage = {
  contentId: string;
  id: string;
  url: string;
  style: TProjectPresentationImageStyle;
  thumbnailUrl: string;
  position?: TTaskPresentationImagePosition;
};

export type TProjectPresentationLink = {
  contentId: string;
  targetId: string;
  targetName: string;
  text: string;
};

export type TProjectPresentationText = {
  type: 'text' | 'header';
  contentId: string;
  value: string;
  style: string;
};

export type TProjectPresentationSlideContent = {
  id: string;
  images: TProjectPresentationImage[];
  links: TProjectPresentationLink[];
  texts: TProjectPresentationText[];
  videos: TProjectPresentationVideo[];
};

export type TProjectPresentationSlide = {
  backgroundColor: string;
  backgroundImage: string;
  content: TProjectPresentationSlideContent;
  description: string;
  id: string;
  iframeUrl: string;
  isShared: boolean;
  name: string;
  notes: string;
  step: number;
  template: TProjectPresentationSlideTemplate;
  subslides: TProjectPresentationSlide[];
  checkInQuestions: TCheckInQuestion[];
  checkInGroups: TCheckInGroup[];
  products: TProduct[];
};

export type TSlideBackgroundImage = {
  id: string;
  thumbnailUrl: string;
  url: string;
};

export type TProjectPresentation = {
  color: string;
  description: string;
  displayName: string;
  id: string;
  name: string;
  slideBackgroundImages: TSlideBackgroundImage[];
  status: ContentStatusesTypes;
  slides: TProjectPresentationSlide[];
  transition: TProjectPresentationSlideTransition;
  typography: TProjectPresentationTypography;
  type: PresentationTypes;
};

export type TUnit = {
  displayName: string;
  id: string;
};

export type TFile = {
  description: string;
  displayName: string;
  id: string;
  url: string;
  step: string;
  filename: string;
};

export type TProjectData = {
  project: TProject;
};

export type TProjectVariables = {
  trackPresentation: boolean;
  shareId: string;
  code: string;
};
