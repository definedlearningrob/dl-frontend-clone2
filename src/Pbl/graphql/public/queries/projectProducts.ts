import { gql } from '@apollo/client';

export default gql`
  query PublicProjectProducts($shareId: ID!, $code: String!) {
    project: task(shareId: $shareId, code: $code) {
      id
      products {
        id
        description
        displayName
        id
        name
        rubrics {
          criteriaLabels {
            displayName
            id
            score
          }
          criterias {
            id
            rubricCriteriaLabelId
            rubricHeadingId
            text
          }
          description
          headings {
            id
            multiplier
            name
          }
          id
          name
          displayName
        }
      }
    }
  }
`;

export type TPublicProjectProductsData = {
  project: {
    id: string;
    products: TProduct[];
  };
};

export type TPublicProjectProductsVariables = {
  shareId: string;
  code: string;
};

export type TProduct = {
  description: string;
  displayName: string;
  id: string;
  name: string;
  rubrics: TRubric[];
  submissionsGradingNeededCount: never;
};

export type TRubric = {
  criteriaLabels: TCriteriaLabel[];
  criterias: TCriteria[];
  description: string;
  displayName: string;
  headings: THeading[];
  pointsAvailable?: number;
  id: string;
  name: string;
};

type TCriteriaLabel = {
  displayName: string;
  id: string;
  score: number;
};

type TCriteria = {
  id: string;
  rubricCriteriaLabelId: string;
  rubricHeadingId: string;
  text: string;
};

type THeading = {
  id: string;
  multiplier: number;
  name: string;
};
