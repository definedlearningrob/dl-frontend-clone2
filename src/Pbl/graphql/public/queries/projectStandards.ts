import { gql } from '@apollo/client';

gql`
  query PublicProjectStandards($shareId: ID!, $setId: String!, $code: String!) {
    project: task(shareId: $shareId, code: $code) {
      id
      standards(setId: $setId) {
        grade
        standardNumber
        standardText
        subject
      }
    }
  }
`;
