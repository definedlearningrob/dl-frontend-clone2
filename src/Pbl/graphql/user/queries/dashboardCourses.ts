import { gql } from '@apollo/client';

import { TRACK_GRADES } from '@shared/resources/constants';
import { ToStringLiteral } from '@shared/utils/types';

export default gql`
  query DashboardCourses($page: Int, $perPage: Int) {
    courses: tracks(page: $page, perPage: $perPage) {
      nodes {
        tasksCount
        description
        displayName
        grades
        id
        imageUrl
        shortDescription
        thumbnailUrl
      }
      nodesCount
      pagesCount
    }
  }
`;

export type TGrade = ToStringLiteral<typeof TRACK_GRADES>;

export type TDashboardCoursesData = {
  courses: TDashboardCoursesTracks;
};

export type TDashboardCoursesVariables = {
  page?: number;
  perPage?: number;
};

export type TDashboardCoursesTracks = {
  nodes: TDashboardCoursesTrack[];
  nodesCount: number;
  pagesCount: number;
};

export type TDashboardCoursesTrack = {
  tasksCount: number;
  description: string;
  grades: TGrade[];
  displayName: string;
  id: string;
  imageUrl: string;
  shortDescription: string;
  thumbnailUrl: string;
};
