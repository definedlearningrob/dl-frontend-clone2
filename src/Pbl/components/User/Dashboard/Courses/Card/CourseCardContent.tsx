/* eslint-disable react/no-danger */
import { isEmpty } from 'lodash-es';
import { useTranslation } from 'react-i18next';

import { TDashboardCoursesTrack, TGrade } from '@pbl/graphql/user/queries/dashboardCourses';

import SharedImage from '@shared/components/Image/Image';
import { cleanInjection } from '@shared/utils/cleanInjection';
import SharedButton from '@shared/components/Button/Button';
import SharedIcon from '@shared/components/Icon/Icon';
import { ReactComponent as ChevronRight } from '@shared/svg/chevron_right.svg';

import { Grades } from './Grades';
import '@pbl/components/User/Dashboard/Courses/Card/Card.sass';

type Props = {
  course: TDashboardCoursesTrack;
  grades: TGrade[];
  onClick: () => void;
};

const CourseCardContent = ({ course, onClick, grades }: Props) => {
  const { displayName, imageUrl, thumbnailUrl, tasksCount, shortDescription } = course;
  const { t } = useTranslation();

  return (
    <div className='dashboard-course-card'>
      <div className='dashboard-course-card__image-wrapper'>
        <SharedImage
          alt={displayName}
          className='dashboard-course-card__image'
          fallbackSrc={imageUrl}
          src={thumbnailUrl}
        />
      </div>
      <div className='card-data' data-testid='card-data'>
        {!isEmpty(grades) && <Grades grades={grades} />}
        <div className='dashboard-course-card__project-summary'>
          <p className='dashboard-course-card__grades-text'>
            {t('user.dashboard.projectsCount', { tasksCount })}
          </p>
        </div>
      </div>
      <div className='dashboard-course-card__content'>
        <h6 className='dashboard-course-card__name' data-testid='dashboard-course-item-name'>
          {displayName}
        </h6>
        <p className='dashboard-course-card__description'>
          <span
            className='dashboard-course-card__short-description-inject'
            dangerouslySetInnerHTML={cleanInjection(shortDescription)}
          />
        </p>
      </div>
      <SharedButton
        className='dashboard-course-card__description-link'
        data-testid='description-link'
        variant='link'
        onClick={onClick}>
        {t('user.dashboard.view')}
        <SharedIcon
          className='dashboard-course-card__description-link-icon'
          icon={<ChevronRight />}
          size='xs'
        />
      </SharedButton>
    </div>
  );
};

export default CourseCardContent;
