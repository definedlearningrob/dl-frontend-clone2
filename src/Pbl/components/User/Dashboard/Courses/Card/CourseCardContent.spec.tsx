import { fireEvent } from '@testing-library/react';

import { renderWithRouter } from '@pbl/utils/test';
import CourseCardContent from '@pbl/components/User/Dashboard/Courses/Card/CourseCardContent';
import { TDashboardCoursesTrack, TGrade } from '@pbl/graphql/user/queries/dashboardCourses';

type Props = {
  course: TDashboardCoursesTrack;
  grades: TGrade[];
  onClick: () => void;
};

const renderComponent = ({ course, onClick, grades }: Props) =>
  renderWithRouter(<CourseCardContent course={course} grades={grades} onClick={onClick} />);

describe('CourseCardContent', () => {
  const mockedCardContent = {
    course: {
      __typename: 'Track',
      tasksCount: 5,
      description: '<p>This course description</p>',
      displayName: 'Test display name',
      grades: [],
      id: '1',
      imageUrl: 'http://localstack.test.me',
      shortDescription: 'short decs',
      thumbnailUrl: 'http://localstack.test.me',
    },
    onClick: jest.fn(),
    grades: ['K', '1', '2', '3', '4'],
  };

  it('should render grades correctly', () => {
    // @ts-ignore
    const { getByText, getByTestId } = renderComponent(mockedCardContent);

    expect(getByTestId('card-data')).toBeInTheDocument();
    expect(getByText('Grades: K-4')).toBeInTheDocument();
  });

  it('should render number of projects correctly', () => {
    // @ts-ignore
    const { getByText } = renderComponent(mockedCardContent);

    expect(getByText(`Projects: ${mockedCardContent.course.tasksCount}`)).toBeInTheDocument();
  });

  it('should fire click event onClick', () => {
    // @ts-ignore
    const { getByTestId } = renderComponent(mockedCardContent);

    fireEvent.click(getByTestId('description-link'));

    expect(mockedCardContent.onClick).toHaveBeenCalled();
  });

  it('should renders card correctly', () => {
    // @ts-ignore
    const { container } = renderComponent(mockedCardContent);

    expect(container).toMatchSnapshot();
  });
});
