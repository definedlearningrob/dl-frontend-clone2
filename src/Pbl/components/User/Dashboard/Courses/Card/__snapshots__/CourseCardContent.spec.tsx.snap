// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseCardContent should renders card correctly 1`] = `
<div>
  <div
    class="dashboard-course-card"
  >
    <div
      class="dashboard-course-card__image-wrapper"
    >
      <img
        alt="Test display name"
        class="dashboard-course-card__image"
        src="http://localstack.test.me"
      />
    </div>
    <div
      class="card-data"
      data-testid="card-data"
    >
      <ul
        class="gradesWrapper"
        data-testid="grades-summary"
      >
        <li
          class="grade"
          data-testid="grade-list-element"
        >
          <p
            class="gradeText"
            data-testid="grade-text"
          >
            Grades: K-4
          </p>
        </li>
      </ul>
      <div
        class="dashboard-course-card__project-summary"
      >
        <p
          class="dashboard-course-card__grades-text"
        >
          Projects: 5
        </p>
      </div>
    </div>
    <div
      class="dashboard-course-card__content"
    >
      <h6
        class="dashboard-course-card__name"
        data-testid="dashboard-course-item-name"
      >
        Test display name
      </h6>
      <p
        class="dashboard-course-card__description"
      >
        <span
          class="dashboard-course-card__short-description-inject"
        >
          short decs
        </span>
      </p>
    </div>
    <button
      class="button flex items-center justify-center gap-xs rounded-sm cursor-pointer text-sm font-medium no-underline focus-visible:outline focus-visible:outline-1 focus-visible:outline-offset-[3px] dashboard-course-card__description-link -link py-xs px-base leading-[22px] text-white"
      data-testid="description-link"
      type="button"
    >
      <span
        class="items-center flex justify-center gap-xs"
      >
        View
        <div
          class="icon dashboard-course-card__description-link-icon -xs-size"
          data-testid="icon"
        >
          <div />
        </div>
      </span>
    </button>
  </div>
</div>
`;
